File: node/http/controllers/subscribeController.js
import express from "express";

const router = express.Router();

export const subscribeController = router.get("/", (_req, res) => {
    try {
        res.json([
            {
                pubsubname: "eventhubs-pubsub",
                topic: "ansalemo-eventhub-topic",
                route: "messages"
            }
        ]);
    } catch (error) {
        console.error(error)
        res.json({ "err": error })
    }
})

File: go/http/controllers/subscribeController.go
package controllers

import (
	"github.com/gofiber/fiber/v2"
)

type Subscription struct {
	PubsubName string            `json:"pubsubname"`
	Topic      string            `json:"topic"`
	Metadata   map[string]string `json:"metadata,omitempty"`
	Route      string            `json:"route"`
}

func SubscribeController(cxt *fiber.Ctx) error {
	subscriptions := []Subscription{
		{
			PubsubName: "pubsub",
			Topic:      "orders",
			Route:      "/orders",
		},
	}

	return cxt.JSON(subscriptions)
}
